cro-clib/src/address.rs

```diff
@@ -0,0 +1,84 @@
+use crate::types::{CroAddressPtr, CroResult};
+use std::ffi::CString;
+use std::ptr;
+/// print address information
+#[no_mangle]
+/// # Safety
+pub unsafe extern "C" fn cro_print_address(
+    address_ptr: CroAddressPtr,
+    buf: *mut u8,
+    buf_length: u32,
+) -> CroResult {
+    let address = address_ptr.as_mut().expect("get address");
+    let address_string = address.address.to_string();
+    if address_string.as_bytes().len() >= buf_length as usize {
+        return CroResult::fail();
+    }
+    ptr::write_bytes(buf, 0, buf_length as usize);
```

Comment:
the names / comments are a bit misleading -- it doesn't "print" address, it copies the textual/printed address into "buf.... perhaps "get_printed_address" instead of "print_address" and "printed_address_output" instead of "buf" + comment for the expected length?

Reply:
i agree, i will change and rebase